dissect/hypervisor/disk/vdi.py

```diff
@@ -67,18 +67,13 @@ def _read(self, offset, length):
 
 
 class Vbox:
-    def __init__(self, fh):
-        self.fh = fh
-        self.xml = ElementTree.fromstring(fh.read())
 
-        self.machine = list(self.xml)[0]
-        self.registry = next((elem for elem in self.machine if "MediaRegistry" in elem.tag))
+    VBOX_XML_NAMESPACE = "{http://www.virtualbox.org/}"
+
+    def __init__(self, fh):
+        xml = ElementTree.fromstring(fh.read())
+        self._disks = xml.findall(f".//{self.VBOX_XML_NAMESPACE}HardDisk[@location][@format='VDI'][@type='Normal']")
```

Comment:
```suggestion
        self.xml = ElementTree.fromstring(fh.read())
```

Reply:
Why would you expose the xml as a public property?

Reply:
I do not understand the rationale behind this suggestion. Can you elaborate?

Reply:
It's neater to not do the pre-reading of the disk attributes in the constructor, if you're not going to use them in the constructor. IMO it makes more sense to only start reading the disk attributes in the actual `disks()` method.